breakerbp/failure_ratio.go

```diff
@@ -76,6 +81,11 @@ type Config struct {
 
 	// Timeout is the duration of the 'Open' state. After an 'Open' timeout duration has passed, the breaker enters 'half-open' state.
 	Timeout time.Duration `yaml:"timeout"`
+
+	// TimeoutJitterRatio is the jitter ratio to be applied to Timeout.
+	//
+	// Optional, default is 0.5, means the actual timeout used can be Timeout+-50%.
+	TimeoutJitterRatio *float64 `yaml:"timeoutJitterRatio"`
```

Comment:
🔕 I don't think this really needs to be a pointer since `0` is normalized by [randbp.JitterRatio](https://github.com/reddit/baseplate.go/blob/4020824b36e41451704c4213f4af376aa65f9015/randbp/jitter.go#L14-L16) to be 1 anyways, we could probably just treat 0 as "not set"?

Reply:
A sane default is important :) If the jitter ratio is (near) 1, we could get to a very very short jittered timeout, which can be bad.

Reply:
To be clear, what I meant was if it's set to `0` we treat that as `not set` and set it to the default value.

Reply:
but comparing float numbers to 0 can be tricky, so pointer is better :)

Reply:
Fair point 😅